### DISCUSSION TITLE
Multivalue fields become one value joined by ';'

### DISCUSSION THREAD

**Discussion by vincentvandercruyssen** [ghDaYuYu/foo_discogs#89] [Help wanted] [answered]
In previous version of the Discogs tagger the values of multivalue fields got separated or split (foobar2000 identified it as multiple values). 
I'm having trouble achieving this with the current version. The semicolon gets printed in between values, but foobar2000 does not seem to interpret it as multiple values. The mapping is exactly the same, in this case $unique(%<RELEASE_LABELS_NAME>%).

Current method:
![foobar2000_rajmbHNLEE](https://user-images.githubusercontent.com/27077333/213861778-2d662fe8-c376-4920-8a94-11b252918775.png)

Previous method, automatically splitting values:
![foobar2000_6t9OooWQzF](https://user-images.githubusercontent.com/27077333/213861791-1f6b5f7d-1cce-4ed6-bfe3-1bffdf1e21f2.png)

**Comment by ghuDaYuYu** [comment]
The latest version of tagger introduces a section in the preference panel: 'Multi-value fields'.
Would you give it a test adding 'Discogs Label' to it as in the following screenshot?

![unique_multi](https://user-images.githubusercontent.com/62167612/214726858-6a8b2352-faa2-470a-801c-555f35c57a76.png)

**Comment by vincentvandercruyssen** [comment]
That section was there previously as well, but it didn't seem to change anything (neither adding it in that field or in the Advanced preferences). I updated to the latest foobar2000 beta 24 and it works now as intended. I am however still on some previous version of foo_discogger 1.0.14.8pv (different fork?), I am on foobar2000 2 x64 version and can't install the component here: https://www.foobar2000.org/components/view/foo_discogs
![foobar2000_35qkISRuK7](https://user-images.githubusercontent.com/27077333/214804723-23f1597d-0726-4f4f-8e39-37ebcc7cec41.png)
![foobar2000_grNhBPKXWx](https://user-images.githubusercontent.com/27077333/214804757-ad2e59e3-8073-4fd9-b40b-2d204d26bed6.png)

**Comment by frogworth** [comment]
I'm also getting these issues with multivalue fields - wrong spacing around semicolons, and on some fields multiple semicolons added at the end (DISCOGS_CREDIT_VOCALS I think).
I'm also on 1.0.14.8pv which is the latest download from Github (October last year) and I'm on the latest v 2.0 beta 24, but on the 32bit version still.

**Comment by ghuDaYuYu** [comment]
@CearUway
> That section was there previously as well, but it didn't seem to change anything (neither adding it in that field or in the Advanced preferences). I updated to the latest foobar2000 beta 24 and it works now as intended. I am however still on some previous version of foo_discogger 1.0.14.8pv (different fork?), I am on foobar2000 2 x64 version and can't install the component here: https://www.foobar2000.org/components/view/foo_discogs

If I am not wrong the 'Multi-value fields' options were added during the v1.0.14.x series with the help of @username116. Unfortunately that account has been deleted, together with the issue documenting the 'multi-value' troubleshooting discussion, hopefully, all pending issues related to multi-values will be fixed in next release iterations.

I am glad that Discogger 1.0.14.8 (preview) on foobar2000 beta 24 is working as intended for you, also that version of Discogger is up to date.  About the official fb2k repository URL, imho, there might never be a x64 foo_discogs released by zoomorph at that location. If foo_discogger ever gets stable enough to be included into the official repository then the URL ending part would be .../view/foo_discogger.

**Reply by username116** [reply]
> the 'Multi-value fields' options were added during the v1.0.14.x series with the help of @username116. Unfortunately that account has been deleted, together with the issue documenting the 'multi-value' troubleshooting discussion

My account was flagged because I was using an email address with a forwarding service. It took a while to get the answer from the support. It's ok now, the discussion (79) has reappeared. Sorry for the inconvenience.

**Reply by ghuDaYuYu** [reply]
Great you managed to bring it back, thanks!

**Comment by ghuDaYuYu** [comment]
@frogworth 
> I'm also getting these issues with multivalue fields - wrong spacing around semicolons, and on some fields multiple semicolons added at the end (DISCOGS_CREDIT_VOCALS I think). I'm also on 1.0.14.8pv which is the latest download from Github (October last year) and I'm on the latest v 2.0 beta 24, but on the 32bit version still.

Thanks for reporting!
About the wrong spacing around semicolons, might that mean that the value is a single value containing semicolons? Did you try including the field into the 'Multi-value fields' or checking the 'Auto-update' related option (before starting the tagging process)?
I would suggest to temporarily turn on the new option 'Log updates' and review the console logs. Also a custom playlist column $meta_sep(DISCOGS_CREDIT_VOCALS, * ) might help troubleshoot further this issue.

**Comment by ghuDaYuYu** [comment]
Here there is a new preview release in the 14.x multi-value fix series...

Change log:
- Fix 'Multi-value fields' preference value user updates (not getting properly applied)
- fb2k SDK and project C++ version update

Disclaimer: Updating to the latest fb2k SDK, and more probably, updating the component to C++ v17, have most probably introduced new bugs unrelated to this release 'multi-values' code updates.

Thanks for testing!

[zipped preview v1.0.14.10.zip](https://github.com/ghDaYuYu/foo_discogs/files/10514239/zipped.preview.v1.0.14.10.zip)

**Comment by frogworth** [comment]
Yay! 1.0.15 fixes my multi-value field issues. Thanks!

**Comment by ghuDaYuYu** [comment, marked as answer]
Nice to hear that!

Just as documentation, here are some special cases...

DISCOGS_ARTISTS_URLS
default:
`%<<ARTISTS_URLS>>%`
multi-value:
`$split($flatten(%<<ARTISTS_URLS>>%))`

DISCOGS_FORMAT
default:
`<default def>`
multi-value:
`$split($join(<default def>))`

Not sure if that is the expected behavior (but it is in my code), any tag definitions producing arrays, ej. `[a, b, c]` when intentionally saved as single values, , will be flatten to `'a;b;c' `
